src/module.c

```diff
@@ -523,31 +523,33 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
         context = aggObject->createContext();
     
     RedisModule_ReplyWithArray(ctx, REDISMODULE_POSTPONED_ARRAY_LEN);
+    int64_t delta_pos = 0;
     while (SeriesIteratorGetNext(&iterator, &sample) != 0 &&
                     (maxResults == -1 || arraylen < maxResults)) {
-        if (aggObject == NULL) { // No aggregation whatssoever
+        // No aggregation 
+        if (aggObject == NULL) { 
             RedisModule_ReplyWithArray(ctx, 2);
-
             RedisModule_ReplyWithLongLong(ctx, sample.timestamp);
             RedisModule_ReplyWithDouble(ctx, sample.value);
             arraylen++;
         } else {
-            timestamp_t current_timestamp = sample.timestamp - (sample.timestamp % time_delta);
```

Comment:
@gkorland and @ashtul the problem of that simple example is here:
since for the initial position we: ` current_timestamp = 5 - 5 % 4 ` = 4 ( when it should be 5.

Reply:
the example we're discussing is:
```
127.0.0.1:6379> TS.ADD a 4 1
(integer) 4
127.0.0.1:6379> TS.ADD a 5 1
(integer) 5
127.0.0.1:6379> TS.ADD a 6 1
(integer) 6
127.0.0.1:6379> TS.ADD a 7 1
(integer) 6
127.0.0.1:6379> TS.ADD a 8 1
(integer) 6
127.0.0.1:6379> TS.RANGE a 5 8 AGGREGATION SUM 4
1) 1) (integer) 4
   2) "3"
2) 1) (integer) 8
   2) "1"
```